### ISSUE TITLE
PTH123 false positive

### ISSUE THREAD

**Issue by JonathanPlasse** [charliermarsh/ruff#2140]
Taken directly from this repository in the `script` folder.

```python
def main(*, name: str, code: str, linter: str) -> None:
    """Generate boilerplate for a new rule."""
    # Create a test fixture.
    with (ROOT_DIR / "resources/test/fixtures" / dir_name(linter) / f"{code}.py").open(
        "a",
    ):
        pass
```

Here is the minimal code

```python
from pathlib import Path

(Path("") / "").open()
```

Generate the error `PTH123` `open("foo")` should be replaced by `Path("foo").open()`

Ruff version: `0.0.231`

**Comment by charliermarsh** [comment]
Definitely an error. Does the originating plugin raise this? (Regardless we should fix.)

**Comment by JonathanPlasse** [comment]
No, it does not.

**Comment by JonathanPlasse** [comment]
The problem comes from `collect_call_path` that returns `["open"]` which is confused for the builtin `open`.
Maybe `collect_call_path` should return `["", "open"]` to separate the two.

**Comment by charliermarsh** [comment]
`collect_call_path` returns `["", "open"]` when `open` is bound to the builtin.

**Comment by charliermarsh** [comment]
I'm guessing the problem here is that we're treating `x.open()` as if it's equivalent to `open` within the rule.

**Comment by charliermarsh** [comment]
I can fix tonight.
